compiler/rustc_resolve/src/macros.rs

```diff
@@ -443,11 +443,16 @@ impl<'a> ResolverExpand for Resolver<'a> {
                 PathResult::NonModule(partial_res) if partial_res.unresolved_segments() == 0 => {
                     return Ok(true);
                 }
+                PathResult::NonModule(..) => {
+                    self.session
+                        .struct_span_err(span, "not sure whether the path is accessible or not")
```

Comment:
Could you add a note saying that the type may potentially have associated items (that's why we are not sure).

Reply:
And also a comment saying that if we get a partially resolved `NonModule` in one namespace, then we should get the same result in any other namespaces, so we can return early.